src/hooks/useStats.tsx

```diff
@@ -67,14 +67,12 @@ const computeStats = (
     { "1d": 0, "7d": 0 }
   );
 
-  const avgTTFB = nodes.reduce((acc, el) => {
-    if (el.ttfbStats["p95_12h"]) {
-      const contrib = el.ttfbStats["p95_12h"] / numberOfNodes;
-      return contrib ? acc + contrib : acc;
-    } else {
-      return acc;
-    }
-  }, 0);
+  const medianTTFBSortedNodes = nodes.sort((a, b) => b.ttfbStats.p95_12h - a.ttfbStats.p95_12h);
+  let medianTTFBLastValidIndex = medianTTFBSortedNodes.findLastIndex(node => Number.isInteger(node.ttfbStats.p95_12h));
+  if (medianTTFBLastValidIndex === -1) {
+    medianTTFBLastValidIndex = 0;
+  }
+  const medianTTFB = medianTTFBSortedNodes[Math.floor((nodes.length - (nodes.length - medianTTFBLastValidIndex)) / 2)]?.ttfbStats?.p95_12h || 0;
```

Comment:
`(nodes.length - (nodes.length - medianTTFBLastValidIndex)) / 2)` could just be `medianTTFBLastValidIndex / 2`